api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+     * Remove an annotation from the specified constructor.
+     *
+     * @param constructor    the constructor to add the annotation to
+     * @param annotationType the annotation to add
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  annotation does not exist on the type or if the constructor is
+     *                                  not currently declared on the type
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructor(AnnotatedConstructor<T> constructor,
+                                                       Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified constructor parameter. If the
+     * constructor is not already present, it will be added. If the constructor
+     * parameter is not already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param position    the position of the parameter to add
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructorParameter(Constructor<T> constructor,
+                                                           int position,
+                                                           Annotation annotation);
+
+    /**
+     * Remove an annotation from the specified constructor parameter.
+     *
+     * @param constructor    the constructor to remove the annotation from
+     * @param position       the position of the parameter to remove
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  constructor is not currently declared on the type or if the
+     *                                  parameter is not declared on the constructor
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructorParameter(Constructor<T> constructor,
+                                                                int position,
+                                                                Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove an annotation from the specified parameter.
+     *
+     * @param parameter      the parameter to remove the annotation from
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  callable which declares the parameter is not currently declared
+     *                                  on the type or if the parameter is not declared on either a
+     *                                  constructor or a method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromParameter(AnnotatedParameter<? super T> parameter,
```

Comment:
Tomas is right that it's not clear what params should this method process. I guess it's intended for both cases - `AnnotatedConstructor` and `AnnotatedMethod`.